common/components/UpdateModal.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { showNotification, TShowNotification } from 'actions/notifications';
+import { Spinner, NewTabLink } from 'components/ui';
+import Modal, { IButton } from 'components/ui/Modal';
+import moment from 'moment';
+import { addListener, sendEvent } from 'utils/electron';
+import { bytesToHuman } from 'utils/formatters';
+import './UpdateModal.scss';
+
+export interface UpdateInfo {
+  version: string;
+  sha512: string;
+  releaseDate: string;
+  releaseName: string;
+  releaseNotes: string;
+}
+
+export interface DownloadProgress {
+  bytesPerSecond: number;
+  percent: number;
+  transferred: number;
+  total: number;
+}
+
+interface Props {
+  isOpen: boolean;
+  updateInfo: UpdateInfo;
+  showNotification: TShowNotification;
+  handleClose(): void;
+}
+
+interface State {
+  isDownloading: boolean;
+  downloadProgress: DownloadProgress | null;
+}
+
+class UpdateModal extends React.Component<Props, State> {
+  public state: State = {
+    isDownloading: false,
+    downloadProgress: null
+  };
+
+  public componentDidMount() {
+    addListener('UPDATE:update-downloaded', () => {
+      sendEvent('UPDATE:quit-and-install');
+    });
+    addListener('UPDATE:download-progress', downloadProgress => {
+      this.setState({ downloadProgress });
+    });
+    addListener('UPDATE:error', err => {
+      console.error('Update failed:', err);
+      this.setState({ isDownloading: false });
+      this.props.showNotification(
+        'danger',
+        <span>
+          Update could not be downloaded, please visit{' '}
+          <NewTabLink href="https://github.com/MyEtherWallet/MyEtherWallet/releases">
+            our github
+          </NewTabLink>{' '}
+          to download the latest release
+        </span>,
+        Infinity
+      );
+    });
+  }
+
+  public render() {
+    const { isOpen, updateInfo, handleClose } = this.props;
+    const { isDownloading, downloadProgress } = this.state;
+    const buttons: IButton[] | undefined = downloadProgress
+      ? undefined
+      : [
+          {
+            text: <span>{isDownloading && <Spinner />} Download Update</span>,
+            type: 'primary',
+            onClick: this.downloadUpdate,
+            disabled: isDownloading
+          },
+          {
+            text: 'Close',
+            type: 'default',
+            onClick: handleClose
+          }
+        ];
+
+    return (
+      <Modal
+        isOpen={isOpen}
+        title={`Update Information`}
+        handleClose={handleClose}
```

Comment:
Should handle canceling the download, if we were downloading. Electron supports this.